srcpkgs/docker/template

```diff
@@ -1,138 +1,10 @@
-
-		SET(CMAKE_FIND_ROOT_PATH  ${XBPS_CROSS_BASE})
-
-		SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-		SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-		SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
-		_EOF
-		cmake_args+=" -DCMAKE_TOOLCHAIN_FILE=cross_${XBPS_CROSS_TRIPLET}.cmake"
-	fi
-	CFLAGS="${CFLAGS/ -pipe / }" CXXFLAGS="${CXXFLAGS/ -pipe / }" \
-	cmake ${cmake_args} ..
-
-	mkdir -p $wrksrc/gopath/src/github.com/docker
-
-	msg_normal "Preparing docker-proxy import path\n"
-	cd $wrksrc/libnetwork-$_libnetwork
-	rm -f $wrksrc/gopath/src/github.com/docker/libnetwork
-	ln -s "$PWD" $wrksrc/gopath/src/github.com/docker/libnetwork
-
-	msg_normal "Preparing docker-cli import path\n"
-	cd $wrksrc/docker-ce-$version
-	rm -f $wrksrc/gopath/src/github.com/docker/cli
-	ln -s "$PWD/components/cli" $wrksrc/gopath/src/github.com/docker/cli
-	rm -f $wrksrc/gopath/src/github.com/docker/docker
-	ln -s "$PWD/components/engine" $wrksrc/gopath/src/github.com/docker/docker
-
-	BUILDTIME=$(date --utc --rfc-3339 ns)
-	BUILDTIME=${BUILDTIME/ /T}
-	vsed -i -e "
-		/GitCommit/s/\".*\"/\"v$version\"/;
-		/Version/s/\".*\"/\"$version\"/;
-		/BuildTime/s/\".*\"/\"$BUILDTIME\"/;
-	" \
-		components/engine/dockerversion/version_lib.go \
-		components/cli/cli/version/version.go
-}
-
-do_build() {
-	export GOPATH="$wrksrc/gopath"
-	mkdir -p $wrksrc/gopath/bin
-
-	msg_normal "Building docker-init\n"
-	cd $wrksrc/tini-$_tini_version/build
-	make tini-static
-
-	msg_normal "Building docker-proxy\n"
-	go build -buildmode=pie \
-		-ldflags="$go_ldflags -linkmode=external" \
-		-o $wrksrc/gopath/bin/docker-proxy \
-		github.com/docker/libnetwork/cmd/proxy
-
-	msg_normal "Building docker cli\n"
-	go build -buildmode=pie \
-		-ldflags="$go_ldflags" -o $wrksrc/gopath/bin/docker \
-		github.com/docker/cli/cmd/docker
-
-	msg_normal "Building docker daemon\n"
-	go build -buildmode=pie \
-		-tags 'libdm_dlsym_deferred_remove seccomp apparmor' \
-		-ldflags="$go_ldflags" \
-		-o $wrksrc/gopath/bin/dockerd \
-		github.com/docker/docker/cmd/dockerd
-}
-
-do_install() {
-	vbin tini-$_tini_version/build/tini-static docker-init
-	vlicense tini-$_tini_version/LICENSE docker-init.MIT
-	vbin gopath/bin/docker-proxy
-	vbin gopath/bin/docker
-	vbin gopath/bin/dockerd
-	cd docker-ce-$version
-	vcompletion components/cli/contrib/completion/bash/docker bash
-	vcompletion components/cli/contrib/completion/zsh/_docker zsh
-	vcompletion components/cli/contrib/completion/fish/docker.fish fish
-	vinstall components/engine/contrib/syntax/vim/syntax/dockerfile.vim 644 \
-		usr/share/vim/vimfiles/syntax/dockerfile.vim
-	vinstall components/engine/contrib/syntax/vim/ftdetect/dockerfile.vim 644 \
-		usr/share/vim/vimfiles/ftdetect/dockerfile.vim
-	vlicense components/engine/contrib/syntax/vim/LICENSE dockerfile.vim.BSD-2-Clause
-	vsv docker
-}
+build_style=meta
+depends="docker-engine docker-cli"
+short_desc="Docker meta-package including the engine and cli tool"
```

Comment:
If `docker-cli` should depend on `moby`, do
```shell
short_desc="Docker container runtime (transitional dummy package)
```
since `docker-cli` would effectively be a complete replacement for the old `docker`.

Otherwise, maybe just "Docker container runtime" to parallel my suggestion for the other two descriptions.

Reply:
I've set it up as a meta-package because docker-cli doesn't explicitly require a local docker daemon, they can be networked.

Reply:
The meta-package is appropriate, but it can still be added as a subpackage of `docker-cli` by putting the following function in your `docker-cli` template:
```shell
docker_package() {
	build_style="meta"
	short_desc="Docker container runtime"
	depends="${sourcepkg}>=${version}_${revision} moby"
}
```
and make `srcpkgs/docker` a symbolic link to `docker-cli`.

Reply:
ahesford: you suggest a restructure of the package to be as follows correct?:

```
docker-cli
 -> docker (transitional package)
moby (docker-engine)
```

this isn't a horrible idea, but I think my approach is better for a few reasons:
1. the docker-cli doesn't depend on the docker container daemon, nor vice versa
2. some people may wish to install one of the two packages individually
3. as it stands, `docker` is a dummy package similar to `xorg` which provides all the tools one might expect or need.
4. renaming the engine package to moby may be more accurate to the repository used, however if someone queries the xbps repositories for docker related packages it might be better to name the package in relation to what it's normally used for. I've also updated the distfiles to be sourced from the `docker/engine` repository so in the event the versioning of these two projects ever differs the package should get the one compatible with other docker tools.

I'd love more input on how to improve things, and I've left responses on your suggested edits.

Reply:
> The meta-package is appropriate, but it can still be added as a subpackage of `docker-cli` by putting the following function in your `docker-cli` template:
> 
> ```shell
> docker_package() {
> 	build_style="meta"
> 	short_desc="Docker container runtime"
> 	depends="${sourcepkg}>=${version}_${revision} moby"
> }
> ```
> 
> and make `srcpkgs/docker` a symbolic link to `docker-cli`.

Implemented.